### ISSUE TITLE
Could this repo solve my problem with Sequelize?

### ISSUE THREAD

**Issue by MichelDiz** [overlookmotel/sequelize-hierarchy#128]
Hi there!

I have this issue https://stackoverflow.com/questions/45175631/relationships-with-mysql-sequelize-and-graphql

It's about hierarchy with an pré-existing Database column.

I'm doing like:

  ```
 tb_user.associate = (models) => {
     tb_user.hasMany(models.ps_user_group, {
       foreignKey: 'id_user',
     });
   };
```

One thing!, I can't put a new table to create relations. Cause this have no compatibility with my CMS.

Take a look on the link.

Best regards.

**Comment by overlookmotel** [comment]
@MichelDiz I'm afraid I don't think this module is what you're looking for. To be honest I don't completely understand what the problem you're trying to solve is - your Stack Overflow question wasn't that clear to me. But I *think* you can do what you're trying to do using just Sequelize - it doesn't sound like a hierarchy to me.

I'm going to close this issue for now, but if you think I'm wrong and you can explain what you're trying to do a bit more clearly, I'll try to help.

**Comment by MichelDiz** [comment]
The thing is I'm trying to do a relation with one "Join table" called "tb_user_group" that has "Id_user" column and an "id_group".

I have a table tb_user that has all information about my users. That has id_user column.

I have tb_group the defines the name of the groups.
That has "id_group" column.

Meaning "tb_user_group" conects the "tb_group" and "tb_user" with each own "ids" columns.

And with sequelize I'm confused what I can do to use to create this kind of relation. "hasmany", "belongsTo", "belongsToMany" and so on.

I appreciate your attention, Thank you.

**Comment by overlookmotel** [comment]
Thanks for clarifying. Yes, you don't need this module (sequelize-hierarchy) to do this.

Sequelize's `.belongsToMany` method is what you're looking for. The `tb_user_group` table acts as the "through" table.

**Comment by MichelDiz** [comment]
Like this? no need to define "foreignKey" right?

```js
const db = {
  tb_user: sequelize.import('./cliente/tb_user'),
  tb_user_group: sequelize.import('./cliente/tb_user_group'),
  tb_group: sequelize.import('./grupos/tb_group'),
};

 db["tb_user"].belongsToMany(db["tb_group"], 
 { through: 'tb_user_group', 
 //foreignKey:'id_user', 
 logging: console.log });

    db["tb_group"].belongsToMany(db["tb_user"], 
    { through: 'tb_user_group', logging: console.log
   // foreignKey:'id_group' 
  });
```

**Comment by overlookmotel** [comment]
@MichelDiz Sorry but Github issues is not the place to ask for help on issues which have nothing to do with the repo that your posting on!

Stack Overflow is the best place. If your post hadn't been so long and complicated, it'd probably have got some responses.

I have tried to point you in the right direction, but I can't help any further.

What you're asking is one of the really basic and standard functionalities of Sequelize. I'd suggest searching the web for an article on how to use Sequelize. I think that will answer all your questions and put you in a good position of understanding.

I'm sorry if this comes across as rude. I know you're just trying to find out the answers to your questions and have no ill intent. But you've got to understand that I, like most people, have finite time and a day job! I have open-sourced this module in the hope that it's useful to people, but I can't act as free tech support helpdesk for anyone who has a question about things which have no relation to the module I've published!
